include/h2/tensor/copy.hpp

```diff
@@ -142,4 +142,61 @@ void Copy(Tensor<DstT>& dst, const Tensor<SrcT>& src)
   }
 }
 
+/**
+ * Return a version of tensor src that is accessible from a device.
+ *
+ * This may return either a copy of the tensor or a view of the
+ * original tensor.
+ *
+ * A view may be returned when the tensor is already on the requested
+ * device; or if the system is a truly unified memory system (like an
+ * APU) where `src`'s device and `dev` share the same physical memory.
+ * In the latter case, the view will have a different device from the
+ * original tensor.
+ *
+ * An optional stream may be provided to control the stream the
+ * returned tensor will be on. If it is not specified, the stream used
+ * will be as follows:
+ * - If `src` is already on `dev`, `src`'s stream will be used.
+ * - Otherwise, `dev`'s default stream will be used.
+ */
+template <typename T>
+std::unique_ptr<Tensor<T>>
+MakeAccessibleOnDevice(Tensor<T>& src,
```

Comment:
```suggestion
make_accessible_on_device(Tensor<T>& src,
```
maybe? we should talk about naming conventions. Because I expect camel case classes, I find that camel-case function names slow down grokking code, but I can cope if that's the consensus.